test/lint/lint-shell.sh

```diff
@@ -27,5 +27,8 @@
 # SC2166: Prefer [ p ] && [ q ] as [ p -a q ] is not well defined.
 # SC2166: Prefer [ p ] || [ q ] as [ p -o q ] is not well defined.
 # SC2181: Check exit code directly with e.g. 'if mycmd;', not indirectly with $?.
-shellcheck -e SC2001,SC2004,SC2005,SC2006,SC2016,SC2028,SC2046,SC2048,SC2066,SC2086,SC2116,SC2148,SC2162,SC2166,SC2181 \
-    $(git ls-files -- "*.sh" | grep -vE 'src/(secp256k1|univalue)/')
+
+docker run -t -v $(pwd):/code -t koalaman/shellcheck-alpine:v0.5.0 \
```

Comment:
NACK this change. Skip in macos for now?

Reply:
@promag I am unsure about what you mean. This part does not have anything to do with macos. I'd happily factor this one out into a separate pull request.

The lint step is executed in linux/trusty and uses a shellcheck that comes with the distro. It kept segfaulting for me (in travis, not on macos), so I am using a different version which runs in docker and has a different (and more modern) executable in it.

This runs on my mac locally as well as in a linux build in travis equally.

Reply:
Please don't introduce a Docker dependency. If `shellcheck` does not run on Mac then just skip it.

Reply:
It does run on Mac. I did not introduce this step because it does not run on mac (it does) – as can be seen in https://github.com/bitcoin/bitcoin/pull/13728 ). For some reason it does segfault though in the `os: linux/dist: trusty` build.

Here is a build where it segfaults: https://travis-ci.org/scravy/bitcoin/jobs/410232096#L623

Reply:
@practicalswift What is the problem with a docker dependency? To me it seems like the shellcheck binary in ubuntu trusty is broken and segfaults under certain conditions. Here is an issue: https://github.com/koalaman/shellcheck/issues/1053 The docker image pulled in is the official build of shellcheck.

For me to understand: Introducing a dependency for example using `brew` on osx - is that okay or not?

Reply:
@scravy With the suggested change everyone running the linter would effectively be vulnerable in case of a compromise of the `koalaman` Docker Hub account.

So a clear NACK from me on this specific change. Try to find a more lightweight workaround that doesn't introduce any additional trust.

Reply:
@MarcoFalke Running all the linters in a docker container sounds intriguing to me. I might tackle that in a separate pull request. Simply to pull in more recent versions.

Reply:
IMO docker should not be a developer dependency and so running linters should also be possible without docker.